tests/unit/blocking_connection_tests.py

```diff
@@ -204,3 +208,33 @@ def test_sleep(self, select_connection_class_mock):
             connection.sleep(0.00001)
 
 
+class _SimpleProxy(object):
+
+    def __init__(self, obj):
+        self._obj = obj
+
+    def __getattribute__(self, item):
+        if item in ('_obj', 'send'):
+            return object.__getattribute__(self, item)
+        return getattr(self._obj, item)
+
+
+class TestChannelCreationOnBrokenConnection(unittest.TestCase):
+
+    def test_that_it_raise_closed_connection(self):
+
+        for error in (errno.EPIPE, errno.ECONNABORTED, errno.ECONNREFUSED):
+            # initiate a connection:
+            conn = blocking_connection.BlockingConnection()
```

Comment:
This relies on actually being able to connect to a live broker. Unit tests are not allowed to do that. All external networking needs to be mocked out as in the other unit tests. Unit tests are expected to work even when the external servers are not running.